app/src/main/java/ru/spbhse/pocketmagic/GestureListener.java

```diff
@@ -0,0 +1,32 @@
+package ru.spbhse.pocketmagic;
+
+import android.gesture.Gesture;
+import android.gesture.GestureLibrary;
+import android.gesture.GestureOverlayView;
+import android.gesture.Prediction;
+
+import java.util.ArrayList;
+
+public class GestureListener implements GestureOverlayView.OnGesturePerformedListener {
+
+    private GestureLibrary gestureLibrary;
+    private GameActivity.Caster caster;
+
+    GestureListener(GestureLibrary gestureLibrary, GameActivity.Caster caster) {
+        this.gestureLibrary = gestureLibrary;
+        this.caster = caster;
+    }
+
+    @Override
+    public void onGesturePerformed(GestureOverlayView gestureOverlayView, Gesture gesture) {
+        ArrayList<Prediction> predictionList = gestureLibrary.recognize(gesture);
+
+        if (predictionList.size() > 0) {
+            Prediction firstPrediction = predictionList.get(0);
+
+            if (firstPrediction.score > 2) {
```

Comment:
What if the score is > 2 for non-first prediction? I this you a for-loop would work better here.

for (Prediction prediction: gestureLibrary.recognize(gesture)) {
  if (prediction.score > 2) {
    caster.cast(prediction.name);
    break;
  }
}

Reply:
We are interested in only the first prediction: the GestureLibrary object returns a list with predictions sorted by `Prediction.score`.